### ISSUE TITLE
x86_64 macOS test suite fails with signal BUS

### ISSUE THREAD

**Issue by tmcgilchrist** [oxcaml/oxcaml#4562]
Prefixing this with I saw "x86 macOS may still work." in the README.md, so perhaps this report is expected. 

On macOS 15.6.1 x86_64, running the `make runtest` fails with:

```
Perm: compose with own inverse: PASSED 1000 attempts (0.007491 s)
File "oxcaml/tests/simd/dune.inc", line 1158, characters 0-136:
1158 | (rule
1159 |  (alias   runtest)
1160 |  (enabled_if (= %{context_name} "main"))
1161 |  (action
1162 |   (with-outputs-to
1163 |    callback.out
1164 |    (run ./callback.exe))))
Command got signal BUS.
File "oxcaml/tests/simd/dune.inc", line 2977, characters 0-156:
2977 | (rule
2978 |  (alias   runtest)
2979 |  (enabled_if (= %{context_name} "main"))
2980 |  (action
2981 |   (with-outputs-to
2982 |    callback_nodynlink.out
2983 |    (run ./callback_nodynlink.exe))))
Command got signal BUS.
make: *** [runtest] Error 1      
```

Loading the executable into llldb
```
$ lldb ./_build/main/oxcaml/tests/simd/callback.exe
(lldb) target create "./_build/main/oxcaml/tests/simd/callback.exe"
Current executable set to '/Users/tsmc/code/ocaml/oxcaml/_build/main/oxcaml/tests/simd/callback.exe' (x86_64).
(lldb) run
Process 95649 launched: '/Users/tsmc/code/ocaml/oxcaml/_build/main/oxcaml/tests/simd/callback.exe' (x86_64)
Process 95649 stopped
* thread #1, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=2, address=0x100c01ff8)
    frame #0: 0x0000000100007bf9 callback.exe`camlCallback__stack_overflow_6_44_code + 9
callback.exe`camlCallback__stack_overflow_6_44_code:
->  0x100007bf9 <+9>:  callq  0x100007bf0    ; <+0>
    0x100007bfe <+14>: addq   $0x8, %rsp
    0x100007c02 <+18>: retq   
    0x100007c03 <+19>: nopw   %cs:(%rax,%rax)
Target 0: (callback.exe) stopped.
(lldb) bt

^C* thread #1, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=2, address=0x100c01ff8)
  * frame #0: 0x0000000100007bf9 callback.exe`camlCallback__stack_overflow_6_44_code + 9
    frame #1: 0x0000000100007bfe callback.exe`camlCallback__stack_overflow_6_44_code + 14
    frame #2: 0x0000000100007bfe callback.exe`camlCallback__stack_overflow_6_44_code + 14
    frame #3: 0x0000000100007bfe callback.exe`camlCallback__stack_overflow_6_44_code + 14
    frame #4: 0x0000000100007bfe callback.exe`camlCallback__stack_overflow_6_44_code + 14
...
```

Printing the backtrace in lldb gets caught in a loop and needs to be interrupted.

CPU is Intel(R) Core(TM) i5-8500B CPU @ 3.00GHz with CPUFLAGS of:

```
FPU VME DE PSE TSC MSR PAE MCE CX8 APIC SEP MTRR PGE MCA CMOV PAT PSE36 CLFSH DS ACPI MMX FXSR SSE SSE2 SS HTT TM PBE SSE3 PCLMULQDQ DTES64 MON DSCPL VMX SMX EST TM2 SSSE3 FMA CX16 TPR PDCM SSE4.1 SSE4.2 x2APIC MOVBE POPCNT AES PCID XSAVE OSXSAVE SEGLIM64 TSCTMR AVX1.0 RDRAND F16C
RDWRFSGS TSC_THREAD_OFFSET SGX BMI1 HLE AVX2 SMEP BMI2 ERMS INVPCID RTM FPU_CSDS MPX RDSEED ADX SMAP CLFSOPT IPT SGXLC MDCLEAR TSXFA IBRS STIBP L1DF ACAPMSR SSBD
SYSCALL XD 1GBPAGE EM64T LAHF LZCNT PREFETCHW RDTSCP TSCI
``` 

Perhaps there is an issue with the simd instructions?

**Comment by tmcgilchrist** [comment]
There is another Bus error in the upstream tests.
```
$ make runtest-upstream
....
List of failed tests:
    tests/asmgen/'catch-float.cmm' with line 4 (asmgen) 
    tests/asmgen/'catch-multiple.cmm' with line 4 (asmgen) 
    tests/asmgen/'catch-rec-deadhandler.cmm' with line 4 (asmgen) 
    tests/asmgen/'catch-rec.cmm' with line 4 (asmgen) 
    tests/asmgen/'catch-try-float.cmm' with line 4 (asmgen) 
    tests/asmgen/'catch-try.cmm' with line 4 (asmgen) 
    tests/asmgen/'even-odd-spill-float.cmm' with line 4 (asmgen) 
    tests/asmgen/'even-odd-spill.cmm' with line 4 (asmgen) 
    tests/asmgen/'even-odd.cmm' with line 4 (asmgen) 
    tests/asmgen/'fib.cmm' with line 4 (asmgen) 
    tests/asmgen/'pgcd.cmm' with line 4 (asmgen) 
    tests/asmgen/'tagged-fib.cmm' with line 4 (asmgen) 
    tests/asmgen/'tagged-tak.cmm' with line 4 (asmgen) 
    tests/asmgen/'tak.cmm' with line 4 (asmgen) 
    tests/misc/'pr7168.ml' with line 8 (native) Process 21152 got signal 10(Bus error
    tests/runtime-errors/'stackoverflow.ml' with line 61 (native) Process 22139 got signal 10(Bus error
    tests/templates/basic/'probe.ml' with line 31 (ocamlopt.byte) 
    tests/typing-layouts-arrays/'test_vec256_u_array.ml' with line 9 (native) 
```

The stop reason in lldb is pointing to a memory protection error:

```
Exception Type:        EXC_BAD_ACCESS (SIGBUS)
Exception Codes:       KERN_PROTECTION_FAILURE at 0x100c01ff8
```

You're probably right @NickBarnes there's something off with the signal handlers.
